### PR TITLE
debug: edited user/admin register

### PR DESCRIPTION
**Pull request by kengthong**
## Description:
- added joi check for register email body to be an email.
- fixed the problem of register returning [object, Object] error message.

## Checklist:
- [ ] Merged latest develop
- [ ] PR title makes sense

### CODE REVIEW

File: src/services/user.service.ts [hunk 1]
```diff
@@ -56,19 +56,17 @@ export default class UserService {
     let errors = [];
```
Comment (ianleck) [inline src/services/user.service.ts]: Can delete this

File: src/services/user.service.ts [hunk 2]
```diff
@@ -56,19 +56,17 @@ export default class UserService {
...
 
     if (!username || !email || !password || !confirmPassword) {
-      errors.push({ msg: 'Please enter all fields' });
+      // errors.push({ msg: 'Please enter all fields' });
```
Comment (ianleck) [inline src/services/user.service.ts]: Just remove these comments since no longer relevant

File: src/services/user.service.ts [hunk 3]
```diff
@@ -56,19 +56,17 @@ export default class UserService {
...
+      throw new Error('Please enter all fields');
     }
 
     if (password != confirmPassword) {
-      errors.push({ msg: 'Passwords do not match' });
+      // errors.push({ msg: 'Passwords do not match' });
```
Comment (ianleck) [inline src/services/user.service.ts]: Same

**Review by ianleck — CHANGES_REQUESTED** [review]

File: src/services/user.service.ts (hunk 3 above)
Comment (kengthong) [reply src/services/user.service.ts]: done

File: src/services/user.service.ts (hunk 2 above)
Comment (kengthong) [reply src/services/user.service.ts]: done

**Review by ianleck — APPROVED** [review]

**Pull request merged**